src/main/java/com/noticemedan/map/viewmodel/MouseController.java

```diff
@@ -1,18 +1,42 @@
 package com.noticemedan.map.viewmodel;
 
+import com.noticemedan.map.model.KDTree.Rect;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
 public class MouseController {
 
-    private final double factor = 1.1;
+    private double zoomLevel = 1.0;
 
-    public void addZoomAbility(CustomPane customPane){
+	/*
+	THIS NEEDS TO BE REDONE TO WORK WITH KD-TREE AND zoomLevel in MapCanvas
+	TODO Make zoomMethod: that pivots around mouseposition.
+	TODO Make zoomMethod: that take use of the zoomLevel variable in MapCanvas.
+	TODO Make zoomMethod: that scales properly without messing with the KD-TREE search area.
+	*/
+	public void addZoomAbility(CustomPane customPane) {
         customPane.getCanvasContent().setOnScroll(event -> {
             event.consume();
 
             if (event.getDeltaY() == 0) return;
 
-            double scaleFactor = (event.getDeltaY() > 0) ? factor : 1 / factor;
+            this.zoomLevel = customPane.getMapCanvas().getZoomLevel();
+            this.zoomLevel = (event.getDeltaY() > 0) ? zoomLevel * 1.1 : zoomLevel * (1 / 1.1);
```

Comment:
Øh giver du zoomLevel en ny værdi to gange her? Hvis det er fordi du skal bruge værdien til den næste udregning, så brug en ny variabel i den første